src/schema/schema.rs

```diff
@@ -40,29 +40,25 @@ pub struct Schema {
 impl Schema {
   /// Gets a nested schema at a certain point.
   pub fn get(&self, mut pointer: Pointer) -> Option<Self> {
-    if pointer.len() == 0 {
+    if pointer.is_empty() {
       Some(self.clone())
     } else {
-      match &self.type_ {
-        &SchemaType::Null => None,
-        &SchemaType::Boolean => None,
-        &SchemaType::Number{..} => None,
-        &SchemaType::String{..} => None,
-        &SchemaType::Array{ref items} => {
+      match self.type_ {
+        SchemaType::Array{ref items} => {
           if INTEGER_RE.is_match(&pointer.remove(0)) {
             items.get(pointer)
           } else {
             None
           }
         },
-        &SchemaType::Object{ref properties,..} => {
+        SchemaType::Object{ref properties,..} => {
           if let Some(schema) = properties.get(&pointer.remove(0)) {
             schema.get(pointer)
           } else {
             None
           }
         },
-        &SchemaType::Enum(_) => None
+        _ => None
```

Comment:
They potentially could, I would like to get the error anyway so I can decide at that time instead of getting a silent error. The pipe solution would be fine although I stylistically prefer splitting it up over a few lines.